logger.go

```diff
@@ -96,3 +112,72 @@ func (l *MatchingLogger) Write(p []byte) (int, error) {
 
 	return len(p), nil
 }
+
+// parsedLog is a lightweight log structure we parse from subsystem logs.
+// Another approach for capturing logs from subsystems is to pass around
+// LogEntry or opentelemetry structs.
+type parsedLog struct {
+	date       []byte
+	level      []byte
+	loggerName []byte
+	location   []byte
+	tail       []byte
+}
+
+func (p parsedLog) valid() bool {
+	return len(p.date) > 0 && len(p.level) > 0 && len(p.loggerName) > 0 && len(p.location) > 0 && len(p.tail) > 0
+}
+
+var levels = map[string]zapcore.Level{
+	"DEBUG":  zapcore.DebugLevel,
+	"INFO":   zapcore.InfoLevel,
+	"WARN":   zapcore.WarnLevel,
+	"ERROR":  zapcore.ErrorLevel,
+	"DPANIC": zapcore.DPanicLevel,
+	"PANIC":  zapcore.PanicLevel,
+	"FATAL":  zapcore.FatalLevel,
+}
+
+var colorCodeRegexp = regexp.MustCompile(`\x1b\[\d+m`)
+
+// stripAnsiColorCodes removes color codes from a string so we can use it internally.
+func stripAnsiColorCodes(raw []byte) []byte {
+	return colorCodeRegexp.ReplaceAll(raw, nil)
+}
+
+// entry converts a parsedLog to a zapcore.Entry which can be NetAppender'd.
+func (p parsedLog) entry() zapcore.Entry {
+	level, ok := levels[string(p.level)]
+	if !ok {
+		level = zapcore.DebugLevel
```

Comment:
An unknown level means the parsing has gone wrong, doesn't it? Feels like that should be Warn or higher to me so it's not masked.

Reply:
done in 1cf86b7d